dss/api/bundles/checkout.py

```diff
@@ -0,0 +1,63 @@
+import json
+
+import boto3
+import os
+import requests
+from flask import jsonify
+from cloud_blobstore import BlobNotFoundError
+
+from dss.util.bundles import get_bundle
+from dss.util.checkout import get_execution_id
+from ... import dss_handler, get_logger
+
+
+@dss_handler
+def post(uuid: str, json_request_body: dict, replica: str, version: str = None):
+
+    sfn = boto3.client('stepfunctions')
+    sts_client = boto3.client("sts")
+
+    email = json_request_body['email']
+
+    if replica is None:
+        replica = "aws"
+
+    try:
+        get_bundle(uuid, replica, version)
+
+        sfn_input = {"bundle": uuid, "version": version, "email": email, "replica": replica}
+        if "destination" in json_request_body:
+            sfn_input["bucket"] = json_request_body["destination"]
+        input = json.dumps(sfn_input)
+
+        region = boto3.Session().region_name
+        accountid = sts_client.get_caller_identity()['Account']
+        stage = os.environ["DSS_DEPLOYMENT_STAGE"]
+        state_machine_arn = f"arn:aws:states:{region}:{accountid}:stateMachine:dss-checkout-sfn-{stage}"
+        response = sfn.start_execution(
+            stateMachineArn=state_machine_arn,
+            name=get_execution_id(),
+            input=input
+        )
+        return jsonify(dict(execution_id=response["executionArn"])), requests.codes.ok
+    except Exception as e:
+        error_message = str(e)
+        return jsonify(dict(code='not_found', title=error_message)), requests.codes.bad
```

Comment:
Please use the canonical name, `requests.codes.bad_request`. Also, catching Exception and then returning it as 400 is wrong. You don't know for sure that the error you caught is the user's fault and not a backend service failure or logic error in the server. Only return 400 series errors when you have enough information to conclude definitively what happened. The catch-all `Exception` catch should default to 500 series.

Reply:
I had difficulties catching specific exceptions as some of the exceptions classes are defined multiple times with the same name. @ttung shared an example of using fully qualified name to avoid the confusion.